src/Type/DefinitionType.ts

```diff
@@ -1,7 +1,16 @@
 import { BaseType } from "./BaseType";
 
 export class DefinitionType extends BaseType {
-    public constructor(private name: string | undefined, private type: BaseType) {
+    public constructor(
+        private name: string | undefined,
+        private type: BaseType,
+        /**
+         * Source Code fileName for DefinitionType. Used at TopRefNodeParser
+         * and ExposeNodeParser. So that, You can override DefinitionFormatter
+         * to custome your self DefinitionType name by fileName.
+         */
+        private sourceFileName?: string
```

Comment:
This doesn't feel clean. Can we make `sourceFileName` an argument to the `DefinitionType` constructor?

_Originally posted by @domoritz in https://github.com/vega/ts-json-schema-generator/pull/1019#discussion_r751336682_